guvnor-services/guvnor-services-backend/src/main/java/org/guvnor/common/services/backend/file/upload/AbstractFileServlet.java

```diff
@@ -294,4 +309,12 @@ protected void processAttachmentDownload( final String url,
         }
     }
 
+    protected String getSessionId( final HttpServletRequest request,
+                                   final SessionProvider sessionProvider ) {
+        final HttpSession session = request.getSession( true );
+        final String clientId = checkNotNull( "clientId", request.getParameter( "clientId" ) );
```

Comment:
Personally I'd prefer a fall-back if ```request.getParamete("clientId")``` returned ```null```. For example default to zero and log a warning that the parameter was not found. The worst it can do is mean the concurrent update mechanism will not work properly. Better than throwing an ```IllegalArgumentException``` IMO.